src/VisualStudio/Core/Def/Venus/ContainedLanguage.cs

```diff
@@ -179,5 +163,24 @@ private void OnDataBufferChanged(object sender, TextContentChangedEventArgs e)
             // when primary buffer has changed to update diagnostic positions.
             _diagnosticAnalyzerService.Reanalyze(this.Workspace, documentIds: SpecializedCollections.SingletonEnumerable(this.ContainedDocument.Id));
         }
+
+        public string GetFilePathFromBuffers()
+        {
+            var textDocumentFactoryService = ComponentModel.GetService<ITextDocumentFactoryService>();
+
+            // Try to get the file path from the secondary buffer
+            if (!textDocumentFactoryService.TryGetTextDocument(SubjectBuffer, out var document))
+            {
+                // Fallback to the primary buffer
+                textDocumentFactoryService.TryGetTextDocument(DataBuffer, out document);
+            }
+
+            if (document == null)
+            {
+                FatalError.ReportAndPropagate(new InvalidOperationException("Failed to get file path for a contained document"));
+            }
+
+            return document?.FilePath ?? string.Empty;
```

Comment:
```suggestion
            return document.FilePath ?? string.Empty;
```

You've already done the null check.

Reply:
Thought that, but the fompiler disagrees:
![image](https://user-images.githubusercontent.com/62502207/178116192-4f07b158-d92a-4ae1-8337-0fe333ab36fe.png)

It seems like it doesn't realize that `FatalError.ReportAndPropagate` will throw the exception and stop execution.

Reply:
@MariaSolOs ReportAndPropagate is assumed to be called in a C# exception filter, so the "and propagate" just means it returns false. It won't actually throw anything at all.

Reply:
@jasonmalinowski makes sense. So I'll just leave the redundant null check then to avoid the compiler warning.